### ISSUE TITLE
Add new top-level field - `webhooks`

### ISSUE THREAD

**Issue by mattpolzin** [mattpolzin/OpenAPIKit#179]
See [release notes](https://github.com/OAI/OpenAPI-Specification/releases/tag/3.1.0-rc0) and [spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.1.0.md), specifically [root object fixed fields](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.1.0.md#fixed-fields).

Work off of the `release/3_0` branch and work on types in the `OpenAPIKit` (not `OpenAPIKit_3_0`) module.

**Comment by mihaelamj** [comment]
@mattpolzin  I put up a PR:
https://github.com/mattpolzin/OpenAPIKit/pull/208

Thanks

**Comment by mihaelamj** [comment]
I encountered a situation where `assertJSONEquivalent` returns `false` for equivalent JSON strings:

`    func test_webhooks_encode() throws {
        let op = OpenAPI.Operation(responses: [:])
        let document = OpenAPI.Document(
          info: .init(title: "API", version: "1.0"),
          servers: [],
          paths: [:],
          webhooks:  [
            "webhook-test": .init(get: op, put: op, post: op, options: op, head: op, patch: op, trace: op)
          ],
          components: .noComponents,
          externalDocs: .init(url: URL(string: "http://google.com")!)
        )
        let encodedDocument = try orderUnstableTestStringFromEncoding(of: document)
        
        let documentJSON: String? =
        """
        {
          "externalDocs": {
            "url": "http:\\/\\/google.com"
          },
          "info": {
            "title": "API",
            "version": "1.0"
          },
          "openapi": "3.0.0",
          "paths": {
          },
          "webhooks": {
            "webhook-test": {
              "delete": {
                "responses": {
                }
              },
              "get": {
                "responses": {
                }
              },
              "head": {
                "responses": {
                }
              },
              "options": {
                "responses": {
                }
              },
              "patch": {
                "responses": {
                }
              },
              "post": {
                "responses": {
                }
              },
              "put": {
                "responses": {
                }
              },
              "trace": {
                "responses": {
                }
              }
            }
          }
        }
        """
      
        assertJSONEquivalent(encodedDocument, documentJSON)
    }`
    
    For that reason I have changed the test to encode, and then decode:
    `  func test_webhooks_encode_decode() throws {
      let op = OpenAPI.Operation(responses: [:])

      let document = OpenAPI.Document(
        info: .init(title: "API", version: "1.0"),
        servers: [],
        paths: [:],
        webhooks:  [
          "webhook-test": .init(get: op, put: op, post: op, options: op, head: op, patch: op, trace: op)
        ],
        components: .noComponents,
        externalDocs: .init(url: URL(string: "http://google.com")!)
      )
    
    // INFO: `assertJSONEquivalent` was returning `false` for equivalent documets, so encode-decode was used for comparison -
      let encodedDocumentString = try orderUnstableTestStringFromEncoding(of: document)
      let encodedDocumentData = (encodedDocumentString?.data(using: .utf8)!)!
      let decodedDocument = try orderUnstableDecode(OpenAPI.Document.self, from: encodedDocumentData)
      XCTAssertEqual(document, decodedDocument)
  }`
  
  However, I don't see that as a perfect solution, for it is obvious that `assertJSONEquivalent` is the culprit and should be addressed.
  
  Should I make a new issue with this in mind?

**Comment by mattpolzin** [comment]
> Should I make a new issue with this in mind?

Yes please! A new issue to track this would be great, especially since you've got a failing example ready to go. I might take a crack at figuring out why the function is not recognizing the two as equivalent this weekend, actually. I hope to find some time for OpenAPIKit today or tomorrow.

**Comment by mattpolzin** [comment]
I took a look at `assertJSONEquivalent` in the context of the failing example you provided. I did not discover a good easy way to fix the helper function, but I did find a way to get your test case passing.

I created a gist here: https://gist.github.com/mattpolzin/b8cf851cb2d61321909ce7d54d3f0a1e

The differences between your example and my gist are:
1. I added the `delete` operation to the `PathItem` `init`.
2. I added a full newline between each empty opening/closing brace of the `responses` property of the serialized JSON.
3. I made sure there was a space between the field names within objects and the following colon ("`:`").

`assertJSONEquivalent` is very unfortunate for how fiddly it is... if you or someone else made it more forgiving, I would be glad to accept changes! The goals of that function are (a) support Linux Foundation's JSON encoding where output differs in the number of spaces given in places and (b) still compare the `String` values of encoded JSON instead of assuming that a full encode/decode cycle captures all possible encode scenarios.

The problem with that function is that it must stick to mostly unmodifying the String values of encoded JSON (exception in Linux where I make what I consider to be an unfortunate compromise) because the only way I can figure to do better without losing anything is by reimplementing some JSON parsing logic to, for example, "remove whitespace between strings without modifying JSON strings" or "remove newlines without modifying JSON strings."
